mycroft/client/speech/service.py

```diff
@@ -125,66 +127,218 @@ def handle_speak(self, event):
 
-    def handle_paired(self, event):
+    def handle_paired(self, message):
         """Update identity information with pairing data.
 
         This is done here to make sure it's only done in a single place.
         TODO: Is there a reason this isn't done directly in the pairing skill?
         """
-        IdentityManager.update(event.data)
+        IdentityManager.update(message.data)
 
-    def handle_audio_start(self, event):
+    def handle_audio_start(self, message):
         """Mute recognizer loop."""
         if self.config.get("listener").get("mute_during_output"):
             self.loop.mute()
 
-    def handle_audio_end(self, event):
+    def handle_audio_end(self, message):
         """Request unmute, if more sources have requested the mic to be muted
         it will remain muted.
         """
         if self.config.get("listener").get("mute_during_output"):
             self.loop.unmute()  # restore
 
-    def handle_stop(self, event):
+    def handle_stop(self, message):
         """Handler for mycroft.stop, i.e. button press."""
         self.loop.force_unmute()
 
+    # stt control bus events
+    def handle_load_stt_lang(self, message):
+        """ tell STT to pre load a model for this language, it will be needed """
+        lang = message.data.get("lang") or Configuration.get().get("lang", "en-us")
+        LOG.info(f"Loading STT lang: {lang}")
+        if hasattr(self.loop.stt, "load_language"):
+            try:
+                self.loop.stt.load_language(lang)
+            except:
+                LOG.exception(f"Failed to load STT lang: {lang}")
+        if hasattr(self.loop.fallback_stt, "load_language"):
+            try:
+                self.loop.fallback_stt.load_language(lang)
+            except:
+                LOG.exception(f"Failed to load fallback STT lang: {lang}")
+
+    def handle_unload_stt_lang(self, message):
+        """ tell STT to unload model for this language, it won't be needed anymore """
+        lang = message.data.get("lang") or Configuration.get().get("lang", "en-us")
+        LOG.info(f"Unloading STT lang: {lang}")
+        if hasattr(self.loop.stt, "unload_language"):
+            try:
+                self.loop.stt.unload_language(lang)
+            except:
+                LOG.exception(f"Failed to unload STT lang: {lang}")
+        if hasattr(self.loop.fallback_stt, "unload_language"):
+            try:
+                self.loop.fallback_stt.unload_language(lang)
+            except:
+                LOG.exception(f"Failed to unload fallback STT lang: {lang}")
+
+    @staticmethod
+    def load_limited_vocabs(message=None):
+        """ loads voc files containing limited vocabulary to be used in STT
+
+        message.data optional parameters:
+
+        lang (str): default to system lang
+        top_words (bool): default True, augment vocab with 10k most common words for lang
+        english_fallback (bool): default False, use en res files if lang file not found
```

Comment:
Would it make more sense to specify an optional fallback language here? English makes sense in many cases, but maybe for Portuguese or Catalan, Spanish would be a more logical default for example?

Reply:
fallback is for failures of main STT, not for extra langs, fallback should be in same lang....

lang being session based the aim is that each engine can support multiple langs, and many of them can, but main stt could for example throw a UnsupporteLang exception. The take away is that each engine should account for the lang being requested inside the plugin itself

Reply:
oops, i misread, you mean voc fallback, i kept this because mycroft does it in their res files.... 

maybe we should remove this, its not like STT would pick it up...

Reply:
I see.. If we keep it, I think it should be a configurable language.. I'm not sure if something like `pt` falling back to `es` would yield usable results (2 most similar languages that come to mind). I have no strong feelings about dropping it though..